external-crates/move/crates/move-binary-format/src/file_format.rs

```diff
@@ -1651,6 +1715,75 @@ pub enum Bytecode {
     ///
     /// ```..., integer_value -> ..., u256_value```
     CastU256,
+    /// Create a variant of the enum type specified via `EnumDefinitionIndex` and push it on the stack.
+    /// The values of the fields of the variant, in the order they appear in the variant declaration,
+    /// must be pushed on the stack. All fields must be provided.
+    ///
+    /// A PackVariant instruction must fully initialize an instance.
+    ///
+    /// Stack transition:
+    ///
+    /// ```..., field(1)_value, field(2)_value, ..., field(n)_value -> ..., variant_value```
+    PackVariant(
+        EnumDefinitionIndex,
+        #[cfg_attr(any(test, feature = "fuzzing"), proptest(strategy = "0u16..255u16"))] VariantTag,
+    ),
+    PackVariantGeneric(
+        EnumDefInstantiationIndex,
+        #[cfg_attr(any(test, feature = "fuzzing"), proptest(strategy = "0u16..255u16"))] VariantTag,
+    ),
+    /// Destroy an variant value of specified by the `VariantTag` that is a member of the enum
+    /// referenced by the `EnumDefinitionIndex` and push the values bound to each variant field on
+    /// the stack.
+    ///
+    /// The values of the fields of the instance appear on the stack in the order defined
+    /// in the enum variant's definition.
+    ///
+    /// This order makes UnpackVariant<T>(tag) the inverse of PackVariant<T>(tag). So
+    /// `UnpackVariant<T>(tag); PackVariant<T>(tag)` is the identity for enum T and variant V with
+    /// tag `t`.
+    ///
+    /// Stack transition:
+    ///
+    /// ```..., instance_value -> ..., field(1)_value, field(2)_value, ..., field(n)_value```
+    UnpackVariant(
+        EnumDefinitionIndex,
```

Comment:
Was wondering if DataTypeHandle index should be used here. In the current implementation, Enum can only be used to help version the internal data structure of a module and cannot be used to represent things like `Option<T>` in rust as there's no external access. Was wondering is `Option<T>` sth you would ever want to support in the future? If so what would be the upgrade path here?